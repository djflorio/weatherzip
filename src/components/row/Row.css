.row {
  margin: 10px;
  padding: 10px;
  color: rgb(0, 92, 104);
}

.row td {
  border-bottom: solid 1px rgb(0, 92, 104);
}

.row__date {
  display: inline-block;
  text-align: center;
  font-weight: 800;
}

.row__date-day {
  margin: 0;
  font-size: 1.3em;
}

.row__date-time {
  margin: 0;
}

.row__description {
  text-transform: capitalize;
  font-weight: 600; 
}

@media (max-width: 450px) {
  .row__date-day {
    font-size: 0.8em;
  }
  
  .row__date-time {
    font-size: 0.6em;
  }

  .row__description {
    font-size: 0.8em;
  }
}