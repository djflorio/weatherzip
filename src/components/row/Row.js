// Third party
import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';

// Assets
import './Row.css';

// Components
import Thermometer from '../thermometer/Thermometer';


const Row = (props) => {

  const day = moment(props.time.split(' ')[0], 'YYYY-MM-DD');
  const parsedDay = day.format('ddd, MMM Do');
  const time = moment(props.time.split(' ')[1], 'HH:mm:ss');
  const parsedTime = time.format('h:mm a');
  
  return (
    <tr className="row">
      <td>
        <div className="row__date">
          <p className="row__date-day">
            {parsedDay}
          </p>
          <p className="row__date-time">
            {parsedTime}
          </p>
        </div>
      </td>
      <td>
        <img className="row__icon" src={props.icon} alt="weather icon" />
      </td>
      <td>
        <p className="row__description">{ props.weather }</p>
      </td>
      <td>
        <Thermometer units={props.units} temp={props.temp} />
      </td>
    </tr>
  );
};

Row.propTypes = {
  units: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
  time: PropTypes.string.isRequired,
  weather: PropTypes.string.isRequired,
  temp: PropTypes.number.isRequired
}

export default Row;