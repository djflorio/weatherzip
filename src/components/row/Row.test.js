import Row from './Row';

const initProps = {
  units: "imperial",
  icon: "icon",
  time: "1776-07-04 12:00:00",
  weather: "Sunny",
  temp: 70
};

function setupShallow(props = initProps) {
  return shallow(<Row {...props} />);
}

it('renders correctly', () => {
  const wrapper = setupShallow();
  expect(wrapper).toMatchSnapshot();
});