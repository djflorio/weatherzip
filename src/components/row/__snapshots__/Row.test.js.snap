// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<tr
  className="row"
>
  <td>
    <div
      className="row__date"
    >
      <p
        className="row__date-day"
      >
        Thu, Jul 4th
      </p>
      <p
        className="row__date-time"
      >
        12:00 pm
      </p>
    </div>
  </td>
  <td>
    <img
      alt="weather icon"
      className="row__icon"
      src="icon"
    />
  </td>
  <td>
    <p
      className="row__description"
    >
      Sunny
    </p>
  </td>
  <td>
    <Thermometer
      temp={70}
      units="imperial"
    />
  </td>
</tr>
`;
